src/components/cluster-deployments/ClusterDeploymentDetailsPage.tsx

```diff
@@ -0,0 +1,172 @@
+import { k8sGet } from '@openshift-console/dynamic-plugin-sdk/api';
+import { AgentClusterInstallKind, AgentKind, ClusterDeploymentKind } from '../../kind';
+import { canEditCluster } from './utils';
+import { SecretModel } from '../../models/ocp';
+
+const { LoadingState, ClusterDeploymentDetails } = CIM;
+
+type DetailsTabProps = React.PropsWithChildren<
+  PageComponentProps<CIM.ClusterDeploymentK8sResource>
+> & {
+  agentClusterInstall: K8sResourceCommon;
+};
+
+const getClusterDeploymentActions =
+  (agentClusterInstall?: CIM.AgentClusterInstallK8sResource): KebabOptionsCreator =>
+  (kindObj: K8sKind, clusterDeployment: K8sResourceCommon) => {
+    const { namespace, name } = clusterDeployment.metadata || {};
+    return [
+      {
+        label: 'Edit',
+        href: `/k8s/ns/${namespace}/${ClusterDeploymentKind}/${name}/edit`,
+        isDisabled: !canEditCluster(agentClusterInstall),
+      },
+    ];
+  };
+
+export const ClusterDeploymentOverview = (props: DetailsTabProps) => {
+  const { obj: clusterDeployment } = props;
+  const [agentClusterInstall, agentClusterInstallLoaded, agentClusterInstallError] =
+    useK8sWatchResource<CIM.AgentClusterInstallK8sResource>(
+      clusterDeployment?.spec?.clusterInstallRef.name
+        ? {
+            kind: AgentClusterInstallKind,
+            name: clusterDeployment?.spec?.clusterInstallRef.name,
+            namespace: clusterDeployment?.metadata?.namespace,
+            namespaced: true,
+            isList: false,
+          }
+        : undefined,
+    );
+
+  const agentSelector = clusterDeployment.spec?.platform?.agentBareMetal?.agentSelector;
+  const [agents, agentsLoaded, agentsError] = useK8sWatchResource<CIM.AgentK8sResource[]>(
+    agentSelector
+      ? {
+          kind: AgentKind,
+          isList: true,
+          selector: agentSelector,
+          namespaced: true,
+        }
+      : undefined,
+  );
+
+  if (!clusterDeployment) return null;
+  if (agentsError) throw new Error(agentsError);
+  if (agentClusterInstallError) throw new Error(agentClusterInstallError);
+  if (!(agentsLoaded && agentClusterInstallLoaded)) return <LoadingState />;
+
+  const handleKubeconfigDownload = async () => {
+    const kubeconfigSecretName =
+      agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    const kubeconfigSecretNamespace = clusterDeployment.metadata.namespace;
+    agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    try {
+      const kubeconfigSecret = await k8sGet(
+        SecretModel,
+        kubeconfigSecretName,
+        kubeconfigSecretNamespace,
+      );
+      const blob = new Blob([atob(kubeconfigSecret.data.kubeconfig)], {
+        type: 'text/plain;charset=utf-8',
+      });
+      saveAs(blob, 'kubeconfig.json');
+    } catch (e) {
+      console.error('Failed to fetch kubeconfig secret.', e);
+    }
+  };
```

Comment:
this should be done inside ai lib.

OCP/ACM needs to provide `k8sGet` implementation with following signature
```
type fetchSecret = (name: string, namespace: string) => Promise<Secret>
```

Reply:
Done